apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py

```diff
@@ -338,7 +345,32 @@ def render(element_html, data):
         )
 
         mcq_options = data["params"][answer_name]
-        mcq_options = filter_multiple_from_array(mcq_options, ["inner_html", "uuid"])
+        mcq_options = filter_multiple_from_array(mcq_options, ["inner_html", "uuid", "tag", "distractor-for"])
+
+        # visual pairing
+        correct_tags = set(block['tag'] for block in mcq_options + student_previous_submission if block['tag'] is not None)
+        incorrect_tags = set(block['distractor-for'] for block in mcq_options + student_previous_submission if block.get('distractor-for', None)) 
+
+        if not incorrect_tags.issubset(correct_tags):
+            raise Exception("The following distractor-for tags do not have matching correct answer tags: " + str(incorrect_tags - correct_tags))
+
+        for block in student_previous_submission + mcq_options:
+            if block.get('distractor-for') is not None:
```

Comment:
Will `"distractor-for"` always be in `block`? If so, you can just use `block["distractor-for"]`, the `.get()` function is only needed if you want to return a default value if the desired key is not in the given dictionary.

Reply:
addressed